osutil/user.go

```diff
@@ -150,14 +151,19 @@ func EnsureUserGroup(name string, id uint32, extraUsers bool) error {
 			delCmdStr = append(delCmdStr, "--extrausers")
 		}
 
-		// TODO: groupdel doesn't currently support --extrausers, so
-		// don't try to clean up when it is specified (LP: #1840375)
-		if !extraUsers {
-			delCmdStr = append(delCmdStr, name)
-			cmd = exec.Command(delCmdStr[0], delCmdStr[1:]...)
-			if output2, err2 := cmd.CombinedOutput(); err2 != nil {
-				return fmt.Errorf("groupdel failed with: %s (after %s)", OutputErr(output2, err2), useraddErrStr)
+		delCmdStr = append(delCmdStr, name)
+		cmd = exec.Command(delCmdStr[0], delCmdStr[1:]...)
+		if output2, err2 := cmd.CombinedOutput(); err2 != nil {
+			var groupdelErrStr string
+			if extraUsers && bytes.Contains(output2, []byte("unrecognized option")) {
```

Comment:
This is tricky - if this code runs localized we will not detect the error. We could always run userdel with "LC_ALL=C.UTF-8" but then the error could be confusing to the user. My feeling is we should just remove the if extraUsers special case and always create the "OutputErr". I.e. something like:
```go
diff --git a/osutil/user.go b/osutil/user.go
index 851d268033..6ebf45e714 100644
--- a/osutil/user.go
+++ b/osutil/user.go
@@ -20,7 +20,6 @@
 package osutil
 
 import (
-       "bytes"
        "fmt"
        "os"
        "os/exec"
@@ -154,13 +153,7 @@ func EnsureUserGroup(name string, id uint32, extraUsers bool) error {
                delCmdStr = append(delCmdStr, name)
                cmd = exec.Command(delCmdStr[0], delCmdStr[1:]...)
                if output2, err2 := cmd.CombinedOutput(); err2 != nil {
-                       var groupdelErrStr string
-                       if extraUsers && bytes.Contains(output2, []byte("unrecognized option")) {
-                               // We have hit https://bugs.launchpad.net/bugs/1840375
-                               groupdelErrStr = "groupdel does not support '--extrausers' option"
-                       } else {
-                               groupdelErrStr = OutputErr(output2, err2).Error()
-                       }
+                       groupdelErrStr := OutputErr(output2, err2).Error()
                        return fmt.Errorf(`errors encountered ensuring user %s exists:
 - %s
 - %s`, name, useraddErrStr, groupdelErrStr)
diff --git a/osutil/user_test.go b/osutil/user_test.go
index 3227670025..996f899137 100644
--- a/osutil/user_test.go
+++ b/osutil/user_test.go
@@ -567,7 +567,7 @@ func (s *ensureUserSuite) TestEnsureUserGroupFailedUseraddCoreNoExtra(c *check.C
        err := osutil.EnsureUserGroup("lakatos", 123456, true)
        c.Assert(err, check.ErrorMatches, `errors encountered ensuring user lakatos exists:
 - useradd failed with: some error
-- groupdel does not support '--extrausers' option`)
+- groupdel: unrecognized option '--extrauser'`)
 
        c.Check(mockGroupDel.Calls(), check.DeepEquals, [][]string{
                {"groupdel", "--extrausers", "lakatos"},
```
and it seems the error is still readable enough(?)

Reply:
Oh, right! I could just search for the `--extrausers` string, which should be printed regardless of the locale, but indeed all this `if` block is a bit too much for just getting a slightly better error message. So I'll happily apply your suggestion :-)

Reply:
Done!